app/domain/authentication/authn_oidc/authenticate_oidc_conjur_token/authenticate_oidc_conjur_token.rb

```diff
@@ -57,11 +57,25 @@ def validate_origin
         end
 
         def audit_success
-          @audit_event.(input: @authenticator_input, success: true, message: nil)
+          @audit_event.(
+            resource_id: @authenticator_input.webservice.resource_id,
+              authenticator_name: @authenticator_input.authenticator_name,
+              account: @authenticator_input.account,
+              username: @authenticator_input.username,
+              success: true,
+              message: nil
+          )
```

Comment:
I'd actually like to remove `authenticator_input` sometime in the future as I don't really like it. It doesn't really tell the story. When we look at a command class it should be fully readable so if i look at the input clause and see `inputs: %i(authenticator_input)` it doesn't tell me anything. However, if i see `inputs: %i(authenticator_name account authenticator_webservice user)` like we did in `validate_status` then I fully understand what the command class needs as an input. I'll add an issue for this refactor and hopefully we'll get to it in the near future.

Regarding this change, in `validate_status` we don't use `authenticator_input` so we can't inject it to `audit_event`

Reply:
Yes, this is a debatable point, but I think if all authenticators take the same inputs (I think this is true, even if some ignore some of the fields), then you in your head you think of it like "a bunch of stuff I need to give to an autheticator."  Whereas if you always are passing the individual fields, you have a

[Data Clump Smell](https://refactoring.guru/smells/data-clumps)

Lmk if I am missing your point though...

Reply:
I see. 

I added a new class `AuthenticatorStatusInput` as it's quite different from what we need in the `AuthenticatorInput`. I aligned it's parameters as much as possible so it will be the same input for `audit_event`.

Having these parameters in one object makes the implementation of the `status` method in the `Authenticator` class look better so there's another plus.